### PR TITLE
Feature: importer metrics collector and fix puppeteer timeout issue

### PR DESCRIPTION
**Pull request by sywhb**
- fix: catch scrapingbee api error
- fix: increase twitter-handler timeout to 60 seconds
- fix: dedupe url if failed to save
- feat: add importer metrics collector
- feat: send email notification once the import has finished
- feat: update matter importer metrics
- feat: add importer metrics collector
- feat: send importer status update
- fix: add start:collector command
- fix: catch importer status update error
- fix: testcases
- add dockerfile

### CODE REVIEW

**Comment by sywhb** [comment]
Fix #2234

**Comment by jacksonh** [comment]
Linking this one: https://github.com/omnivore-app/omnivore/issues/2301

**Review by jacksonh — APPROVED** [review]
This is really good. Thank you!

File: packages/puppeteer-parse/index.js [hunk 1]
```diff
@@ -379,14 +410,21 @@ async function fetchContent(req, res) {
...
     if (context) {
       await context.close();
     }
     console.info(`parse-page`, logRecord);
-  }
 
-  return res.sendStatus(200);
+    // send import status to update the metrics
+    if (taskId) {
+      await sendImportStatusUpdate(userId, taskId, importStatus);
```
Comment (jacksonh) [inline packages/puppeteer-parse/index.js:423]: clever!

**Comment by sywhb** [comment]
Let me make the required changes in GCP and merge it!

File: packages/import-handler/src/luaScripts/updateMetrics.lua [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  -- get the total, imported and failed counters
+  local result = {}
+	local nextkey
+	for i, v in ipairs(bulk) do
+		if i % 2 == 1 then
+			nextkey = v
+		else
+			result[nextkey] = v
+		end
+	end
```
Comment (Nezteb) [inline packages/import-handler/src/luaScripts/updateMetrics.lua:20]: Should probably install https://github.com/prettier/plugin-lua 😄 

Also: why is the Lua needed? Could this update not be done in TS?

Comment (sywhb) [reply packages/import-handler/src/luaScripts/updateMetrics.lua:20]: Thanks @Nezteb , `prettier` is always needed for every repo and the Lua script here is to guarantee the operation is atomic

**Pull request merged**